app/services/articles/feeds/weighted_query_strategy.rb

```diff
@@ -186,10 +186,14 @@ class WeightedQueryStrategy
               AND comments.deleted = false
               AND comments.created_at > :oldest_published_at"]
         },
-        # Weight to give for the number of intersecting tags the given
-        # user follows and the article has.
-        matching_tags_factor: {
-          clause: "LEAST(10.0, SUM(followed_tags.points))::integer",
+        # Weight to give for the the intersection of positive tag follows and the articles tags.  We
+        # look at the sum of the followed points.
+        matching_positive_tag_intersection_sum_of_points_factor: {
+          enabled: false,
+          # NOTE: here we're using points and not explicit points.  There does exist drift from the
+          # explicit points (e.g. what the user set for the tag) and what we've calculated based on
+          # their interactions (e.g. the `points`).
+          clause: "GREATEST(0.0, LEAST(10.0, SUM(followed_tags.points)))::integer",
```

Comment:
min/max logic always surprises me. The original idea was "limit the weight of matched article tags to 10.0 max", and is extended here to "eliminate the negative tags from the match"?